di-ipv-ticf-stub/lambdas/src/management/handlers/managementHandler.ts

```diff
@@ -14,51 +14,49 @@ export async function handler(
   if (!userId) {
     return buildApiResponse({ errorMessage: "Missing userId." }, 400);
   }
-  const statusCode = event.pathParameters?.statusCode ?? "200";
-  let ticfManagementRequest: TicfManagementRequest;
-  try {
-    ticfManagementRequest = parseRequest(event);
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  } catch (error: any) {
-    console.error(error);
-    return buildApiResponse({ errorMessage: error.message }, 400);
-  }
 
   try {
+    const ticfManagementRequest = parseAndValidateRequest(event);
+
     await persistUserEvidence(
       decodeURIComponent(userId),
       ticfManagementRequest,
-      parseInt(statusCode),
     );
-    return buildApiResponse({ message: "Success !!" });
+    return buildApiResponse({ message: "Success!!" });
+
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
   } catch (error: any) {
     console.error(error);
-    return buildApiResponse({ errorMessage: error.message }, 500);
+    return buildApiResponse(
+      { errorMessage: error.message },
+      error.statusCode || 500,
+    );
   }
 }
 
-function parseRequest(event: APIGatewayProxyEventV2): TicfEvidenceItem {
-  if (!event.pathParameters?.statusCode && !event.body) {
-    throw new Error("Missing request body");
+function parseAndValidateRequest(
+  event: APIGatewayProxyEventV2,
+): TicfManagementRequest {
+  if (!event.body) {
+    throw new BadRequestError("Missing request body");
   }
 
-  let ticfManagementRequest: TicfManagementRequest = {};
-  if (event.body) {
-    ticfManagementRequest = JSON.parse(event.body);
+  const ticfManagementRequest: TicfManagementRequest = JSON.parse(event.body);
 
-    if (ticfManagementRequest && !ticfManagementRequest.type) {
-      throw new Error("Invalid request");
-    }
+  if (
+    !ticfManagementRequest.evidence?.type &&
+    (!ticfManagementRequest.statusCode || ticfManagementRequest.statusCode == 200)
+  ) {
+    throw new BadRequestError("Invalid request - evidence type required if configuring a successful TICF response");
```

Comment:
Just covering edge cases since both `statusCode` and `evidence` are optional but you're right, probably a case that we don't need to verify. I did leave a check for if an evidence is provided, it should contain a `type` to make sure someone can't just do something like `{evidence: "not-the-right-thing"}`

Reply:
Would be cleaner with a schema validator like yup but seems a bit much for this 😅

Reply:
Yeah - you could do some `typeof` checks (to make sure it's an object) but it's only a stub, so let's not worry too much!